src/containers/permissions/DataSetPermissionsContainer.js

```diff
@@ -0,0 +1,212 @@
+/*
+ * @flow
+ */
+
+import React, {
+  useEffect,
+  useMemo,
+  useReducer,
+  useState,
+} from 'react';
+
+import { List, Map, Set } from 'immutable';
+import { PaginationToolbar, Typography } from 'lattice-ui-kit';
+import { ReduxUtils, useRequestState } from 'lattice-utils';
+import { useDispatch, useSelector } from 'react-redux';
+import { RequestStates } from 'redux-reqseq';
+import type {
+  Ace,
+  EntitySet,
+  PermissionType,
+  Principal,
+  UUID,
+} from 'lattice';
+import type { RequestState } from 'redux-reqseq';
+
+import DataSetPermissionsCard from './DataSetPermissionsCard';
+
+import { Spinner, StackGrid } from '../../components';
+import {
+  GET_PAGE_DATA_SET_PERMISSIONS,
+  INITIALIZE_DATA_SET_PERMISSIONS,
+  getPageDataSetPermissions,
+  initializeDataSetPermissions,
+} from '../../core/permissions/actions';
+import { PERMISSIONS } from '../../core/redux/constants';
+import {
+  selectAtlasDataSets,
+  selectEntitySets,
+  selectOrganizationAtlasDataSetIds,
+  selectOrganizationEntitySetIds,
+  selectPermissionsByPrincipal,
+} from '../../core/redux/selectors';
+import { getDataSetField } from '../../utils';
+import { INITIAL_PAGINATION_STATE, PAGE, paginationReducer } from '../../utils/stateReducers/pagination';
+import type { DataSetPermissionTypeSelection } from '../../types';
+import type { State as PaginationState } from '../../utils/stateReducers/pagination';
+
+const MAX_PER_PAGE = 10;
+
+const { reduceRequestStates } = ReduxUtils;
+
+const DataSetPermissionsContainer = ({
+  filterByPermissionTypes,
+  filterByQuery,
+  onSelect,
+  organizationId,
+  principal,
+  selection,
+} :{|
+  filterByPermissionTypes :Array<PermissionType>;
+  filterByQuery :string;
+  onSelect :(selection :?DataSetPermissionTypeSelection) => void;
+  organizationId :UUID;
+  principal :Principal;
+  selection :?DataSetPermissionTypeSelection;
+|}) => {
+
+  const dispatch = useDispatch();
+
+  const [openDataSetId, setOpenDataSetId] = useState('');
+  const [paginationState, paginationDispatch] = useReducer(paginationReducer, INITIAL_PAGINATION_STATE);
+  const { page, start } = paginationState;
+
+  const getPermissionsRS :?RequestState = useRequestState([PERMISSIONS, GET_PAGE_DATA_SET_PERMISSIONS]);
+  const initializePermissionsRS :?RequestState = useRequestState([PERMISSIONS, INITIALIZE_DATA_SET_PERMISSIONS]);
+
+  const atlasDataSetIds :Set<UUID> = useSelector(selectOrganizationAtlasDataSetIds(organizationId));
+  const atlasDataSets :Map<UUID, Map> = useSelector(selectAtlasDataSets(atlasDataSetIds));
+  const entitySetIds :Set<UUID> = useSelector(selectOrganizationEntitySetIds(organizationId));
+  const entitySets :Map<UUID, EntitySet> = useSelector(selectEntitySets(entitySetIds));
+  const atlasDataSetIdsHash :number = atlasDataSetIds.hashCode();
+  const atlasDataSetsHash :number = atlasDataSets.hashCode();
+  const entitySetIdsHash :number = entitySetIds.hashCode();
+  const entitySetsHash :number = entitySets.hashCode();
```

Comment:
we could avoid using this hashCode hack if the entitySets were already grouped by organizationId in the redux store.

Reply:
I realized I don't need to call `hashCode()` on the ids; updated. To update the data sets would be a larger change, better suited as a follow-up pr.